go/tasks/plugins/k8s/sagemaker/sagemaker.go

```diff
@@ -0,0 +1,320 @@
+	taskTemplate, err := taskCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to fetch task specification")
+	} else if taskTemplate == nil {
+		return nil, errors.Errorf("nil task specification")
+	}
+
+	taskInput, err := taskCtx.InputReader().Get(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to fetch task inputs")
+	}
+
+	// Get inputs from literals
+	inputLiterals := taskInput.GetLiterals()
+	trainPathLiteral, ok := inputLiterals["train"]
+	if !ok {
+		return nil, errors.Errorf("train input not specified")
+	}
+	validatePathLiteral, ok := inputLiterals["validation"]
+	if !ok {
+		return nil, errors.Errorf("validation input not specified")
+	}
+	staticHyperparamsLiteral, ok := inputLiterals["static_hyperparameters"]
+	if !ok {
+		return nil, errors.Errorf("static hyperparameters input not specified")
+	}
+	tunableHyperparamsLiteral, ok := inputLiterals["tunable_hyperparameters"]
+	// TODO: consider not erroring out when this happens, because it could be the case that the user doesn't want
+	// 		 to do HPO
+	if !ok {
+		return nil, errors.Errorf("tunable hyperparameters input not specified")
+	}
+
+	outputPath := createOutputPath(taskCtx.OutputWriter().GetOutputPrefixPath().String())
+	sagemakerJob := smIdl.SagemakerHPOJob{}
+	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &sagemakerJob)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid task specification for taskType [%s]", sagemakerTaskType)
+	}
+
+	// Convert the hyperparameters to the spec value
+	staticHyperparams, err := convertStaticHyperparamsLiteralToSpecType(staticHyperparamsLiteral)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not convert static hyperparameters to spec type")
+	}
+
+	// If the container is part of the task template you can access it here
+	container := taskTemplate.GetContainer()
+
+	// When adding env vars there are some default env vars that are available, you can pass them through
+	envVars := flytek8s.DecorateEnvVars(ctx, flytek8s.ToK8sEnvVar(container.GetEnv()), taskCtx.TaskExecutionMetadata().GetTaskExecutionID())
+	_ = envVars
+
+	taskName := taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetID().NodeExecutionId.GetExecutionId().GetName()
+
+	hpoJob := &hpojobv1.HyperparameterTuningJob{
+		Spec: hpojobv1.HyperparameterTuningJobSpec{
+			HyperParameterTuningJobName: &taskName,
+			HyperParameterTuningJobConfig: &commonv1.HyperParameterTuningJobConfig{
+				ResourceLimits: &commonv1.ResourceLimits{
+					MaxNumberOfTrainingJobs: ToInt64Ptr(10),
+					MaxParallelTrainingJobs: ToInt64Ptr(5),
+				},
+				Strategy: "Bayesian",
+				HyperParameterTuningJobObjective: &commonv1.HyperParameterTuningJobObjective{
+					Type:       "Minimize",
+					MetricName: ToStringPtr("validation:error"),
+				},
+				// TODO: need to implement a factory that map names to different types of parameter range
+				ParameterRanges: &commonv1.ParameterRanges{
```

Comment:
this is not using your params?

Reply:
I hadn't pushed my changes when you commented